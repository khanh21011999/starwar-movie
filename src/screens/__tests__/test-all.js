import React from 'react'
import MovieList from '../movie-list'
import { shallow } from 'enzyme'
import { fireEvent, render, waitFor } from '@testing-library/react-native'
import MovieDetail from '../movie-detail'
import CharacterDetail from '../character-detail'
describe('Test movie l', () => {
	test('Test movie list', () => {
		const props = {}
		const tree = render(<MovieList {...props} />)
		expect(tree.toJSON()).toMatchSnapshot()
		// const testTouch = tree.getByRole('TouchableOpacity')
	   // fireEvent.press(testTouch)
		expect(tree.queryByTestId('header')).toBeTruthy()
	
	})

	// NOT WORK YET vector icon expo issue in 0.69
	// test('Movie Detail', () => {
	// 	const tree = render(<MovieDetail />)
	// 	expect(tree.toJSON()).toMatchSnapshot()
	// })
	// test('Character Detail', () => {
	// 	const tree = render(<CharacterDetail />)
	// 	expect(tree.toJSON()).toMatchSnapshot()
	// })
})
