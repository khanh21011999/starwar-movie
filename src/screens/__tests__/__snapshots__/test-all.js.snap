// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test movie l Test movie list 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "padding": 16,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 53,
        "fontWeight": "bold",
      }
    }
    testID="header"
  >
    Star Wars Movie List
  </Text>
  <View>
    <ActivityIndicator
      size="small"
    />
  </View>
</RCTSafeAreaView>
`;
